symengine/utilities/tests/test_code_quality.py

```diff
@@ -0,0 +1,62 @@
+from os import walk, pardir
+from os.path import split, join, abspath, exists
+
+### CHANGE THIS IF THIS FILE IS EVER MOVED 
+### RIGHT NOW IT'S IN symengine/symengine/utilities/tests/
+SYMENGINE_PATH = abspath(join(split(__file__)[0], pardir, pardir))
+BIN_PATH = join(abspath(join(SYMENGINE_PATH, pardir)), "bin")
+assert exists(SYMENGINE_PATH)
+
+class Check:
+    """
+    A class which defines a particular 'check'.
+
+    Arguments
+    =========
+    checker_fx          : [function] takes in absolute filepath as argument and does the checking
+    root_dirs           : [list] directories on which the check will run recursively
+    included_filetypes  : [optional] [list] patterns, the file will be matched against
+    excluded_names      : [optional] [list] file won't be checked, if it's abs path is `in` excluded_names
+    """
+
+    def __init__(self, checker_fx, root_dirs, included_filetypes=["."], excluded_names=[]):
+        self.fx = checker_fx
+        self.root_dirs = root_dirs
+        self.included_filetypes = included_filetypes
+        self.excluded_names = excluded_names
+
+# Construct your check classes here
+all_checks = []
+errors = False
+
+# trailing whitespace check
+def trailing_whitespace_function(file_path):
+    global errors
+    with open(file_path) as file_ptr:
+        for line_num, line in enumerate(file_ptr):
+            if line.endswith(" \n") or line.endswith("\t\n"):
+                print "Trailing whitespace :", file_path, str(line_num + 1)
+                errors = True
+
+trailing_whitespace_check = Check(trailing_whitespace_function, [SYMENGINE_PATH])
+trailing_whitespace_check.excluded_names = ["/build/", "/doc/", "/cmake/", "/utilities"]
```

Comment:
We have `conf.py` in `/doc/` . Add a test for that too , incase that file is changed in the future.